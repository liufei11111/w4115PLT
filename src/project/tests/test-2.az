Int i;
i = 2;
Boolean b;
b = True;

Int main(Int argc, String argv) {
	Structure s = { a = 1, b = 2, c = 3 };
    Matrix m(1,1); 
    m3 = (m +. m2) * 4;
    Option o = {a = 1};
    o -> a;
    argc = 	s -> a;
    if (i == 2 || i) {
		return argc;
	} else {
		return i+1;
	}
	while ( i >= 2 || (i < 100 && i != 3)) 
		argv = "argv";
	for (i = 0; i < 5; i = i + 1) {
		i = i * (5 + i);
	}
}

Void main2(Int argc2, Matrix m) {
	main(a, "str");
}

