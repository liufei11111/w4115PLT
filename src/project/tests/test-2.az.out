stmts:

int i;
i = 2;
bool b;
b = True;

fdecls:

int main(int argc, string argv)
{
Structure s = {a = 1, b = 2, c = 3};
matrix m(1, 1);
m3 = ( m +. m2 ) * 4;
Option o = {a = 1};
o->a;
argc = s->a;
if (i == 2 || i) {
  return argc;
}
else {
  return i + 1;
}
while (i >= 2 || ( i < 100 && i != 3 )) argv = "argv";
for (i = 0 ; i < 5 ; i = i + 1) {
  i = i * ( 5 + i );
}
}

void main2(int argc2, matrix m)
{
main(a, "str");
}


